packages/headless/src/api/search/search-api-error-response.ts

```diff
@@ -21,9 +22,29 @@ function buildDisconnectedError(
   };
 }
 
-export function buildAPIResponseFromErrorOrThrow(error: Error): {
+function buildIgnoredAbortedError(
+  error: DOMException
+): SearchAPIErrorWithStatusCode {
+  return {
+    statusCode: error.code,
+    type: error.name,
+    message: error.message,
+    ignore: true,
+  };
+}
+
+export function buildAPIResponseFromErrorOrThrow(
+  error: Error,
+  disableAbortWarning?: boolean
+): {
   error: SearchAPIErrorWithStatusCode;
 } {
+  const asDOMException = error as DOMException;
+  if (disableAbortWarning && asDOMException.code === 20) {
```

Comment:
Instead of casting `Error` to `DOMException`, would it be possible to change the parameter type from `Error` to `Error | DOMException` and then do this?
```suggestion
  if (disableAbortWarning && asDOMException?.code === 20) {
```

I haven't tested this, I don't actually know if that's something Typescript allows.

Reply:
Also, I just checked, and `code` seems to be deprecated.
This might be better:
```suggestion
  if (disableAbortWarning && error?.name === 'AbortError') {
```